packages/jwt-verifier/test/spec/verifier.spec.js

```diff
@@ -214,10 +218,215 @@ describe('Jwt Verifier', () => {
+      const claims = {
+        aud: 'http://myapp.com/',
+        iss: ISSUER,
+      };
+
+      const token = new njwt.Jwt(claims)
+        .setSigningAlgorithm('RS256')
+        .setSigningKey(rsaKeyPair.private)
+        .setHeader('kid', rsaKeyPair.public) // For override of key retrieval below
+        .compact();
+
+      const verifier = new OktaJwtVerifier({
+        issuer: ISSUER,
+        testing: {
+          disableHttpsCheck: OKTA_TESTING_DISABLEHTTPSCHECK
+        }
+      });
+      mockKidAsKeyFetch(verifier);
+
+      return verifier.verifyAccessToken(token, 'http://myapp.com/');
+    });
+
+    it('passes when given an audience matching expectation array', () => {
+      const claims = {
+        aud: 'http://myapp.com/',
+        iss: ISSUER,
+      };
+
+      const token = new njwt.Jwt(claims)
+        .setSigningAlgorithm('RS256')
+        .setSigningKey(rsaKeyPair.private)
+        .setHeader('kid', rsaKeyPair.public) // For override of key retrieval below
+        .compact();
+
+      const verifier = new OktaJwtVerifier({
+        issuer: ISSUER,
+        testing: {
+          disableHttpsCheck: OKTA_TESTING_DISABLEHTTPSCHECK
+        }
+      });
+      mockKidAsKeyFetch(verifier);
+
+      return verifier.verifyAccessToken(token, [ 'one', 'http://myapp.com/', 'three'] );
+    });
+
+    it('fails with a invalid audience when given a valid expectation', () => {
+      const claims = {
+        aud: 'http://wrong-aud.com/',
+        iss: ISSUER,
+      };
+
+      const token = new njwt.Jwt(claims)
+        .setSigningAlgorithm('RS256')
+        .setSigningKey(rsaKeyPair.private)
+        .setHeader('kid', rsaKeyPair.public) // For override of key retrieval below
+        .compact();
+
+      const verifier = new OktaJwtVerifier({
+        issuer: ISSUER,
+        testing: {
+          disableHttpsCheck: OKTA_TESTING_DISABLEHTTPSCHECK
+        }
+      });
+      mockKidAsKeyFetch(verifier);
+
+      return verifier.verifyAccessToken(token, 'http://myapp.com/')
+        .then( () => { throw new Error('Invalid audience claim was accepted') } )
+        .catch(err => {
+          expect(err.message).toBe(`audience claim http://wrong-aud.com/ does not match expected audience: http://myapp.com/`);
+        });
+    });
+
+    it('fails with a invalid audience when given an array of expectations', () => {
+      const claims = {
+        aud: 'http://wrong-aud.com/',
+        iss: ISSUER,
+      };
+
+      const token = new njwt.Jwt(claims)
+        .setSigningAlgorithm('RS256')
+        .setSigningKey(rsaKeyPair.private)
+        .setHeader('kid', rsaKeyPair.public) // For override of key retrieval below
+        .compact();
+
+      const verifier = new OktaJwtVerifier({
+        issuer: ISSUER,
+        testing: {
+          disableHttpsCheck: OKTA_TESTING_DISABLEHTTPSCHECK
+        }
+      });
+      mockKidAsKeyFetch(verifier);
+
+      return verifier.verifyAccessToken(token, ['one', 'http://myapp.com/', 'three'])
```

Comment:
What happens if i do `verifyAccessToken(token, [])` ? Seems like  test we should have

Reply:
What happens is a guaranteed failure, since the actual audience won't (can't) match any entry in the empty array, but it's a reasonable enough test to add.

Reply:
I'd advocate for adding a test, so that in the future we can refactor internals and know we're catching this case.  I've been bitten by internal logical that allows an empty value to bypass validations 😬

Reply:
Done